pkg/template/nstemplatetier_test.go

```diff
@@ -0,0 +1,214 @@
+package template_test
+
+import (
+	"testing"
+
+	"github.com/codeready-toolchain/api/pkg/apis"
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/member-operator/pkg/template"
+	"github.com/codeready-toolchain/toolchain-common/pkg/cluster"
+
+	"github.com/operator-framework/operator-sdk/pkg/log/zap"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/h2non/gock.v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	"sigs.k8s.io/kubefed/pkg/apis/core/common"
+	"sigs.k8s.io/kubefed/pkg/apis/core/v1beta1"
+)
+
+func TestGetNSTemplateTier(t *testing.T) {
+
+	// given
+	// Setup Scheme for all resources
+	err := apis.AddToScheme(scheme.Scheme)
+	require.NoError(t, err)
+	logf.SetLogger(zap.Logger())
+	kubeFedCluster, secret := newHostCluster("host")
+	basicTier := &toolchainv1alpha1.NSTemplateTier{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "toolchain.dev.openshift.com/v1alpha1",
+			Kind:       "NSTemplateTier",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "basic",
+			Namespace: "toolchain-host-operator",
+		},
+	}
+	cl := fake.NewFakeClient(secret, basicTier)
+	service := cluster.KubeFedClusterService{Log: logf.Log, Client: cl}
+	service.AddKubeFedCluster(kubeFedCluster)
+
+	t.Run("success", func(t *testing.T) {
+		// when
+		tmpls, err := template.GetNSTemplates("basic")
+
+		// then
+		require.NoError(t, err)
+		require.Len(t, tmpls, 3)
+	})
+
+	t.Run("failures", func(t *testing.T) {
+
+		t.Run("unknown tier", func(t *testing.T) {
+			// when
+			_, err := template.GetNSTemplates("unknown")
+			// then
+			require.Error(t, err)
+			assert.Contains(t, err.Error(), "unable to retrieve the NSTemplateTier 'unknown' from 'Host' cluster")
+		})
+	})
+
+}
+
+func newHostCluster(name string) (*v1beta1.KubeFedCluster, *corev1.Secret) {
+	// make sure that the client gets a valid 200 OK response
+	// when trying to ping the `/api` endpoint on the fake host cluster
+	gock.Observe(gock.DumpRequest)
+
+	gock.New("https://" + name).
```

Comment:
May I ask you why did you choose this approach of using gock mocks instead of the fake client?

Reply:
removed in https://github.com/codeready-toolchain/member-operator/commit/21bb803d4a1d43875aabaf0f26d516f94a817ad2 after our fruitful discussion ;)